docs/source/guide/s3-example-creating-buckets.rst

```diff
@@ -27,6 +27,22 @@ The name of an Amazon S3 bucket must be unique across all regions of the AWS
 platform. The bucket can be located in a specific region to minimize latency
 or to address regulatory requirements.
 
+.. warning::
+
+    You need to make sure that the region in your aws configuration matches the
+    region where you want your bucket to be created. By default, s3 will create
+    a bucket in ``us-east-1`` if the ``LocationConstraint`` is not specified.
+    However, if the region in your configuration doesn't match the
+    ``LocationConstraint``, you'll get ``IllegalLocationConstraintException``
+    error when calling ``create_bucket`` function.
+
+    To avoid this error, you should either:
+
+    * Option 1: Make sure that your ``LocationConstraint`` is same as the region in your
+      configuration, or
+    * Option 2: If you're using a default value for ``LocationConstraint``, make sure
+      that the region in your configuration is set to ``us-east-1``.
+
 .. code-block:: python
```

Comment:
nit - Since we're already touching this file, can you fix the formatting of the following line from the code example. If should be updated from:
```
s3_client.create_bucket(Bucket=bucket_name,
                                    CreateBucketConfiguration=location)
```
to
```
s3_client.create_bucket(
    Bucket=bucket_name, CreateBucketConfiguration=location
)
```

Reply:
I thought adding a warning presents a bigger perspective of the issue to the user, so I added a warning instead of modifying the example. At the same time I agree that modifying the example, makes it simpler for the user to follow. I'll edit the example like you said.